sympy/tensor/indexed.py

```diff
@@ -359,7 +259,10 @@ def args(self):
             return self._args
 
     def _hashable_content(self):
-        return Expr._hashable_content(self) + (self._shape,)
+        if self._shape:
+            return super(IndexedBase, self)._hashable_content() + (self._shape,)
+        else:
+            return super(IndexedBase, self)._hashable_content()
```

Comment:
The issue is that you can make the comparison `None > None` (which raises an exception) in line 192 of `core/basic.py`, within method `compare`:

    c = (l > r) - (l < r)

So as it stands `_hashable_content` should not return a tuple that includes `None`.

I didn't think this would be controversial which is why I just bundled it in. I will try to break things up more in the future though.